bundles/org.openhab.binding.dirigera/src/main/java/org/openhab/binding/dirigera/internal/handler/light/ColorLightHandler.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.dirigera.internal.handler.light;
+
+import static org.openhab.binding.dirigera.internal.Constants.*;
+
+import java.util.Iterator;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.json.JSONObject;
+import org.openhab.binding.dirigera.internal.interfaces.Model;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.HSBType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link ColorLightHandler} for lights with hue, saturation and brightness
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class ColorLightHandler extends TemperatureLightHandler {
+    private final Logger logger = LoggerFactory.getLogger(ColorLightHandler.class);
+
+    private HSBType hsbStateReflection = new HSBType(); // proxy to reflect state to end user
+    private HSBType hsbDevice = new HSBType(); // strictly holding values which were received via update
+
+    public ColorLightHandler(Thing thing, Map<String, String> mapping) {
+        super(thing, mapping);
+        super.setChildHandler(this);
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        if (super.checkHandler()) {
+            JSONObject values = gateway().api().readDevice(config.id);
+            handleUpdate(values);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        super.handleCommand(channelUID, command);
+        String channel = channelUID.getIdWithoutGroup();
+        if (CHANNEL_LIGHT_COLOR.equals(channel)) {
+            if (command instanceof HSBType hsb) {
+                // respect sequence
+                switch (lightConfig.fadeSequence) {
+                    case 0:
+                        brightnessCommand(hsb);
```

Comment:
This confirms my hypothesis that the API would have two sub-channels namely `color` (the `HS` part) and brightness (the `B` part). So IMHO your approach is good.